doc/storage/flash_map: Move to FIXED_PARTITION_ API (#50942)
zephyrproject-rtos/zephyr: doc/services/storage/flash_map/flash_map.rst

The ``<storage/flash_map.h>`` API allows accessing information about device
flash partitions via :c:struct:`flash_area` structures.

Each :c:struct:`flash_area` describes a flash partition. The API provides access
to a "flash map", which contains predefined flash areas accessible via globally
unique ID numbers. The map is created from "fixed-partition" compatible entries

> in DTS file. User may also create :c:struct:`flash_area`  objects at runtime

Reviewer: ```suggestion
in DTS file. Users may also create :c:struct:`flash_area` objects at runtime
```

for application-specific purposes.

This documentation uses "flash area" when referencing single "fixed-partition"

> entity.

Reviewer: ```suggestion
entities.
```

> The :c:struct:`flash_area` contains pointer to :c:struct:`device`,

Reviewer: ```suggestion
The :c:struct:`flash_area` contains a pointer to a :c:struct:`device`,
```

> which can be used to access flash device, an area is placed on, directly

Reviewer: ```suggestion
which can be used to access the flash device an area is placed on directly
```

with the :ref:`flash API <flash_api>`.
Each flash area is characterized by a device it is placed on, offset
from the beginning of the device and size on the device.

> Additional parameter is identifier which is used by :c:func:`flash_area_open`

Reviewer: ```suggestion
An additional identifier parameter is used by the :c:func:`flash_area_open`
```

function to find flash area in flash map.

The flash_map.h API provides functions for operating on a :c:struct:`flash_area`.
The main examples are :c:func:`flash_area_read` and :c:func:`flash_area_write`.
These functions are basically wrappers around the flash API with additional

> offset and size checks, to limit flash operations to predefined area.

Reviewer: ```suggestion
offset and size checks, to limit flash operations to a predefined area.
```

> Most of Flash Map API functions require :c:struct:`flash_area` object pointer
> characterizing flash area they will be working on. There are two possible
> methods to obtain such pointer:

Reviewer: ```suggestion
Most ``<storage/flash_map.h>`` API functions require a :c:struct:`flash_area` object pointer
characterizing the flash area they will be working on. There are two possible
methods to obtain such a pointer:
```

 * obtain it using `flash_area_open`;

>  * defining :c:struct:`flash_area` type object, which requires providing
>    valid :c:struct:`device` object pointer, and offset and size of the area

Reviewer: ```suggestion
 * defining a :c:struct:`flash_area` type object, which requires providing
   a valid :c:struct:`device` object pointer with offset and size of the area
```

   within the flash device.

:c:func:`flash_area_open` uses numeric identifiers to search flash map for

> :c:struct:`flash_area` objects, and returns, if found, pointer to object

Reviewer: ```suggestion
:c:struct:`flash_area` objects and returns, if found, a pointer to an object
```

   :language: DTS
   :start-after: start-after-here

> Partition offset shall is expressed in relation to the flash memory beginning

Reviewer 2: shall is> shall be

.. _MCUboot documentation: https://mcuboot.com/

Numeric flash area ID is obtained by passing DTS node label to
:c:macro:`FIXED_PARTITION_ID()`; for example to obtain ID number
for ``slot0_partition``, user would invoke ``FIXED_PARITION_ID(slot0_partition)``.

All :c:macro:`FIXED_PARTITION_` macros take DTS node labels as partition
identifiers.

> User does not have to obtain :c:struct:`flash_area` object pointer,
> using :c:func:`flash_map_open`, to get information on flash area size, offset
> or device, if such area is defined in DTS file. Knowing DTS node label
> of an area, user may use :c:macro:`FIXED_PARTITION_OFFSET()`,

Reviewer: ```suggestion
Users do not have to obtain a :c:struct:`flash_area` object pointer
using :c:func:`flash_map_open` to get information on flash area size, offset
or device, if such area is defined in DTS file. Knowing the DTS node label
of an area, users may use :c:macro:`FIXED_PARTITION_OFFSET()`,
```

:c:macro:`FIXED_PARTITION_SIZE()` or :c:macro:`FIXED_PARTITION_DEVICE()`
respectively to obtain such information directly from DTS node definition.
For example to obtain offset of ``storage_partition`` it is enough to
invoke ``FIXED_PARTITION_OFFSET(storage_partition)``.

> Below example shows how to obtain :c:struct:`flash_area` object pointer

Reviewer: ```suggestion
Below example shows how to obtain a :c:struct:`flash_area` object pointer
```
